tests/meltano/core/plugin/singer/test_catalog.py

```diff
@@ -1098,6 +1098,166 @@ def test_node_selection(self, node: dict, selection_type: SelectionType) -> None
         """Test that selection metadata produces the expected selection type member."""
         assert ListSelectedExecutor.node_selection(node) == selection_type
 
+    def test_select_metadata_rules_stream_property_creates_both_rules(self) -> None:
```

Comment:
These tests pass if I restore `catalog.py` to the `main` version, so I think it's worth reverting 33bc2017b867ef3020272ec9a4f5d78650c64b50 now that we want to actually target v4.